### DISCUSSION TITLE
iMessage and FaceTime issue

### DISCUSSION THREAD

**Discussion by thedeadfish59** [chris1111/Wireless-USB-Big-Sur-Adapter#300] [Show and tell]
Hi, my iMessage works when I use iPhone USB tethering, but it gets broken when I use Cudy USB Wifi WU1400 (Realtek) with your utility app, though thanks for this project! Just suggest me how I can fix it.
Note: its hackintosh and all are configured as it should be, and as its working with USB tethering, so serial and other stuffs are not something i guess which causing the issue. 

Also, AirPlay or AirDrop works with this one by doing anything?

**Comment by chris1111** [comment]
It does not work with Apple services, it is a known issue 🧐

**Reply by thedeadfish59** [reply]
Understood, fact is, I tried changing mac address with random one, and with USB wifi, iMessages started working, but after a reboot, it again got broken, thus I thought you will know something, thanks for replying.

**Reply by chris1111** [reply]
Apple service work only with Broadcom and Wifi `itlwm` The bad thing of Realtek macOS drivers is there is no source code exist to upgrade this utility.

**Reply by thedeadfish59** [reply]
This is sad indeed, however, I made it working in a sort hacky weird way, I created a virtual interface and added Ethernet as well as usb wifi interface, and all apple services are functional now.

![image](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/assets/10620753/cd2b658d-30ec-496b-b55c-15afdfefe03a)

**Reply by AlluraRed** [reply]
Mate you're a lifesaver. I didn't know imessage wasn't working because you need a supported wifi card lol i spent WEEKS on this smfh. I only have ethernet. What USB wifi do you use? I was thinking of getting a TP Link Archer T3U.

**Comment by chris1111** [comment]
> This is sad indeed, however, I made it working in a sort hacky weird way, I created a virtual interface and added Ethernet as well as usb wifi interface, and all apple services are functional now.
> 
> ![image](https://private-user-images.githubusercontent.com/10620753/294721901-cd2b658d-30ec-496b-b55c-15afdfefe03a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ2MDQyMjAsIm5iZiI6MTcwNDYwMzkyMCwicGF0aCI6Ii8xMDYyMDc1My8yOTQ3MjE5MDEtY2QyYjY1OGQtMzBlYy00OTZiLWI1NWMtMTVhZmRmZWZlMDNhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA3VDA1MDUyMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTMzMDgzOTBkMGUyODQ2YWFhMDEzZWRhN2E1ZTQxYWZiMGNmMjZlZmI5MGRmNGIzMzlhNTcxOTAzOTc4ZjY1N2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._jJFoMCBkLH8tzVhjEjYe3lGEuB2iiK3MvM6hC53p5U)


Do you please make a short video about this for other Hack user
Thanks

**Reply by thedeadfish59** [reply]
Sure, I will be glad to be a help! Thanks Chris for your project. I will reply here once I have created the video.

**Reply by thedeadfish59** [reply]
Hi, I have created a video regarding how I fixed it! 
https://youtu.be/xPyW-DwQha8

Side Note: You may find in video that Wireless Utility Signal bars as White instead of Blue, that's because I modified those icns!

**Reply by chris1111** [reply]
Thanks I try and it work but I have to unplug the Ethernet cable to make it work
 I will ad you video link to the fix page

**Comment by chris1111** [comment]
I create a [New Page](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/blob/master/Fix_Apple_Services.md) for the Fix

**Reply by thedeadfish59** [reply]
Cheers! I liked how well you make a detailed guide with icon and formatting!

**Reply by chris1111** [reply]
Thanks to you! 
all credit is from you 🫵🏿

**Comment by chris1111** [comment]
It work also [SIP activated ](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/discussions/159) ![2303232987](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/assets/6248794/3eaadb85-db29-45f0-a00f-bea93a75342e)

![2464933482](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/assets/6248794/3809db5c-5941-49ad-8a17-95b6782e2601)

![Screenshot 2024-01-07 at 9 09 41 PM](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/assets/6248794/710d21ec-8d9d-48d7-ae04-1a1df98cfb3a)

![Screenshot 2024-01-07 at 9 03 22 PM](https://github.com/chris1111/Wireless-USB-Big-Sur-Adapter/assets/6248794/9b782600-1e36-4f4c-b9d6-e4e5d38c0a12)

**Comment by AlluraRed** [comment]
i can log into imessage then it signs me out 2-3 seconds later. facetime doesn't work either, everything else works.

i only use ethernet and got no compatible wifi adapters. could that be the issue?

**Comment by seragrub14** [comment]
Please help guys. I followed the guide: Unplugged my ethernet cable, setup the interface. But if try to search something up for example I have no connection. (but connected in the top right corner). The iP`s are also switching weirdly in the bride it self. I can't get a working wifi connection using the bridge. But if I plug the ethernet cable in, it works like normal (with the bridge) 

bringe and ethernet: 
![Bildschirmfoto 2024-12-27 um 07 26 12](https://github.com/user-attachments/assets/1a0a4dbe-0e78-4cca-a8e7-97753981c142)

bridge and wifi: 
![Bildschirmfoto 2024-12-27 um 07 28 11](https://github.com/user-attachments/assets/430e9701-48e7-4bee-a65e-62e15a54e89d)

**Reply by chris1111** [reply]
bridge is yellow, it seem to be ok

**Reply by seragrub14** [reply]
yes i also thought so but i get no connection to the internet neither to the router. Like in the pictures the ips are changing weirdly.

**Reply by chris1111** [reply]
Watch
[video
](https://www.youtube.com/watch?v=xPyW-DwQha8)

**Reply by seragrub14** [reply]
Yeah like I Said didn't help. But I fixed it. Just delete the ethernet service and create a bridge with only wifi. 

https://github.com/user-attachments/assets/a9857b5e-7863-444c-bf30-2ebbc1de4fe8
